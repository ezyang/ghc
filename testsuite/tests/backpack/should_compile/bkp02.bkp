unit p where
    signature H where
        data T
    module A where
        import H
        data A = MkA T

unit q where
    module H where
        data T = T

unit r where
    include q
    include p
    module R where
        import A
        import H
        x = MkA T
