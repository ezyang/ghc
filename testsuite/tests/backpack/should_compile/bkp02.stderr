[1 of 3] Processing p
unit p(hole:H)
  [1 of 2] Compiling H[sig]           ( p/H.hsig, nothing )
  [2 of 2] Compiling A                ( p/A.hs, nothing )
[2 of 3] Processing q
  Instantiating q
unit q
  [1 of 1] Compiling H                ( q/H.hs, bkp02-out/q/H.o )
[3 of 3] Processing r
  Instantiating r
unit r
- include q [(H, q:H)]
- include p(q:H) [(A, p(q:H):A)]
  [1 of 2] Including q
  [2 of 2] Including p
    Instantiating p(q:H)
unit p(q:H)
    [1 of 2] Compiling H[sig]           ( p/H.hsig, nothing )
    [2 of 2] Compiling A                ( p/A.hs, bkp02-out/p/p-D5Mg3foBSCrDbQDKH4WGSG/A.o )
  [1 of 1] Compiling R                ( r/R.hs, bkp02-out/r/R.o )
